lib/gringotts/gateways/pin_pay.ex

```diff
@@ -0,0 +1,423 @@
+          + You could use the same config or update it the with your "secrets"
+          as described [above](#module-registering-your-monei-account-at-PinPay).
+
+  2. Run an `iex` session with `iex -S mix` and add some variable bindings and
+  aliases to it (to save some time):
+  ```
+  iex> alias Gringotts.{Response, CreditCard, Gateways.Pinpay}
+  iex> card = %CreditCard{first_name: "Jo",
+                          last_name: "Doe",
+                          number: "4200000000000000",
+                          year: 2099, month: 12,
+                          verification_code: "123", brand: "VISA"}
+  ```
+
+  > Add any other frequently used bindings up here.
+
+  We'll be using these in the examples below.
+
+  [gs]: https://github.com/aviabird/gringotts/wiki/
+  [home]: https://pinpayments.com
+  [docs]: https://pinpayments.com/developers/api-reference
+  [example]: https://github.com/aviabird/gringotts_example
+  """
+
+  # The Base module has the (abstract) public API, and some utility
+  # implementations.  
+  use Gringotts.Gateways.Base
+
+  # The Adapter module provides the `validate_config/1`
+  # Add the keys that must be present in the Application config in the
+  # `required_config` list
+  use Gringotts.Adapter, required_config: []
+
+  import Poison, only: [decode: 1]
+
+  alias Gringotts.{Money, CreditCard, Response}
+
+  @test_url "https://test-api.pinpayments.com/1/"
+  @production_url "https://api.pinpayments.com/1/"
+
+  @doc """
+  Performs a (pre) Authorize operation.
+
+  The authorization validates the `card` details with the banking network,
+  places a hold on the transaction `amount` in the customer’s issuing bank.
+
+  > ** You could perhaps:**
+  > 1. describe what are the important fields in the Response struct
+  > 2. mention what a merchant can do with these important fields (ex:
+  > `capture/3`, etc.)
+
+  PINPAY returns a **Payment Id** (available in the `Response.authorization`
+  field) which can be used later to:
+  * `capture/3` an amount.
+  * `refund/3` the amount.
+
+  ## Optional Fields
+      options[
+        email_id: String,
+        description: String,
+  ip_address: String (optional)     
+      ]
+
+
+
+
+  ## Example
+
+  The following example shows how one would (pre) authorize a payment of $20 on
+  a sample `card`.
+  ```
+  iex> card = %CreditCard{first_name: "Jo",
+                          last_name: "Doe",
+                          number: "4200000000000000",
+                          year: 2099, month: 12,
+                          verification_code: "123", brand: "VISA"}
+  iex> money = %{value: Decimal.new(20), currency: "USD"}
+  iex> {:ok, auth_result} = Gringotts.authorize(Gringotts.Gateways.Pinpay, money, card)
+  ```
+  """
+
+  @spec authorize(Money.t(), CreditCard.t(), keyword) :: {:ok | :error, Response}
+  def authorize(amount, %CreditCard{} = card, opts) do
+    {currency, value, _} = Money.to_integer(amount)
```

Comment:
You docs say that PinPayments does not process money in cents. Shouldn't you be calling `Money.to_string/1` instead?

Reply:
We tried that earlier, but error code 422  with the message `"Amount must be an integer"` was returned!

Reply:
So how do you make PinPayments process a payment of $3.14? Send `314` as `amount`? If yes, that implies PinPayments works in _cents_ not dollars.